etl/grapher_helpers.py

```diff
@@ -212,24 +214,38 @@ def _expand_jinja(obj: Any, dim_dict: Dict[str, str]) -> Any:
         return obj
 
 
-def _title_column_and_dimensions(title: str, dims: List[str], dim_names: List[str]) -> str:
+def _title_column_and_dimensions(title: str, dim_dict: Dict[str, Any]) -> str:
     """Create new title from column title and dimensions.
     For instance `Deaths`, ["age", "sex"], ["10-18", "male"] will be converted into
     Deaths - Age: 10-18 - Sex: male
     """
-    dims = [f"{dim_name.capitalize()}: {dim}" for dim, dim_name in zip(dims, dim_names)]
-
+    dims = [f"{dim_name.replace('_', ' ').capitalize()}: {dim_value}" for dim_name, dim_value in dim_dict.items()]
     return " - ".join([title] + dims)
 
 
-def _underscore_column_and_dimensions(column: str, dims: List[str], dim_names: List[str]) -> str:
+def _underscore_column_and_dimensions(column: str, dim_dict: Dict[str, Any], trim_long_short_name: bool = True) -> str:
     # add dimension names to dimensions
-    dims = [f"{dim_name}_{dim}" for dim, dim_name in zip(dims, dim_names)]
+    dims = [f"{dim_name}_{dim_value}" for dim_name, dim_value in dim_dict.items()]
 
     # underscore dimensions and append them using double underscores
     # NOTE: `column` has been already underscored in a table
     slug = "__".join([column] + [underscore(n) for n in dims])
-    return cast(str, slug)
+    short_name = cast(str, slug)
+
+    if len(short_name) > 255:
+        if trim_long_short_name:
+            unique_hash = f"_{checksum_str(short_name)}"
+            short_name = short_name[: (255 - len(unique_hash))] + unique_hash
+            log.warning(
+                "short_name_trimmed",
+                short_name=short_name,
+                column=column,
+                dims=dim_dict,
+            )
+        else:
+            raise AssertionError(f"short_name {short_name} is too long for MySQL variables.shortName column")
+
+    return short_name
```

Comment:
This makes total sense, and keeps us within a helpful max char length for `short_name`.

It's clear that what we're packing into `short_name` is not that helpful though. I wonder what alternatives there are for us in the long term. I'll bring it to a `#data-architecture` meeting.

Reply:
Cool, perhaps we can come up with a better solution. This happens extremely rarely, though, and only for massive datasets like WDI or GHO. I don't think it's worth spending too much time on it.